docker/runtime/nodejs/kubeless.js

```diff
@@ -18,89 +23,114 @@ const funcHandler = process.env.FUNC_HANDLER;
+const libPath = path.join(modRootPath, 'node_modules');
+const pkgPath = path.join(modRootPath, 'package.json');
+const libDeps = helper.readDependencies(pkgPath);
+
+const { timeHistogram, callsCounter, errorsCounter } = helper.prepareStatistics('method', client);
 helper.routeLivenessProbe(app);
 helper.routeMetrics(app, client);
-const functionCallingCode = `
-try {
-  Promise.resolve(module.exports.${funcHandler}(event, context)).then((result) => {
-    switch(typeof result) {
-      case 'string':
-        _res.end(result);
-        break;
-      case 'object':
-        _res.json(result);
-        break;
-      default:
-        _res.end(JSON.stringify(result))
-    }
-    _end();
-  }).catch((err) => {
-    // Catch asynchronous errors
-    _handleError(err);
-  });
-} catch (err) {
-  // Catch synchronous errors
-  _handleError(err);
-}`;
-const { vmscript, sandbox } = helper.loadFunc(modName, functionCallingCode);
 
-app.all('*', (req, res) => {
-  res.header('Access-Control-Allow-Origin', '*');
-  if (req.method === 'OPTIONS') {
-    // CORS preflight support (Allow any method or header requested)
-    res.header('Access-Control-Allow-Methods', req.headers['access-control-request-method']);
-    res.header('Access-Control-Allow-Headers', req.headers['access-control-request-headers'])
-    res.end();
-  } else {
-    const funcLabel = modName + "-" + req.method;
-    const end = statistics.timeHistogram.labels(funcLabel).startTimer();
-    statistics.callsCounter.labels(funcLabel).inc();
-    const handleError = (err) => {
-      statistics.errorsCounter.labels(funcLabel).inc();
-      res.status(500).send('Internal Server Error');
-      console.error(`Function failed to execute: ${err.stack}`);
-    };
-    let data = req.body.toString('utf-8');
-    if (req.get('content-type') === 'application/json') {
-      data = JSON.parse(data)
-    }
-    const reqSandbox = Object.assign({
-      event: {
-        data,
-        'event-type': req.get('event-type'),
-        'event-id': req.get('event-id'),
-        'event-time': req.get('event-time'),
-        'event-namespace': req.get('event-namespace'),
-        extensions: {
-          request: req,
-        },
-      },
-      context: {
-        'function-name': funcHandler,
-        timeout,
-        runtime: process.env.FUNC_RUNTIME,
-        'memory-limit': process.env.FUNC_MEMORY_LIMIT,
-      },
-      _handleError: handleError,
-      _res: res,
-      _end: end,
-      process: Object.assign({}, process),
-    }, sandbox);
+const script = new vm.Script(fs.readFileSync(modPath), {
+    filename: modPath,
+    displayErrors: true,
+});
+
+function modRequire(p, req, res, end) {
+    if (p.indexOf('./') === 0)
+        return require(path.join(path.dirname(modPath), p));
+    else if (libDeps.includes(p))
+        return require(path.join(libPath, p));
+    else if (p === 'kubeless')
+        return (handler) => modExecute(handler, req, res, end);
```

Comment:
here is where we could add the possibility of importing different functions. What I have tried:

```
    else if (p === 'kubeless')
        return (handler) => {
            let func = null
            if (typeof handler === "function") {
                func = handler;
            } else if (typeof handler === "object") {
                func = handler[funcHandler];
            } else {
                throw new Error(`Unable to load ${handler}`);
            }
            modExecute(handler, req, res, end)
        };
```

Reply:
Yes, would work. Map could also be supported. Better check also for null, e.g.

`} else if (handler && typeof handler === "object") {`

Just a question: if there are multiple functions in one file, would also allow to run it in the same pod, e.g. take the FUNC_NAME from request and use it as key? Or is it just about about developers convenience to use one file.

Reply:
And maybe put `(p === 'kubeless')` to the begin to override any kubeless from functions package.json.

Reply:
> Just a question: if there are multiple functions in one file, would kubeless also allow to run it in the same pod, e.g. take the FUNC_NAME from request and use it as key? Or is it just about developers convenience to use one file.

The second, it is just for convenience but only one function will run per pod.